memgpt/schemas/document.py

```diff
@@ -0,0 +1,14 @@
+import uuid
+from typing import Dict, Optional
+
+from pydantic import BaseModel, Field
+
+
+class DocumentModel(BaseModel):
+    """Representation of a single document (broken up into `Passage` objects)"""
+
+    user_id: uuid.UUID = Field(..., description="The unique identifier of the user associated with the document.")
+    text: str = Field(..., description="The text of the document.")
+    data_source: str = Field(..., description="The data source of the document.")
+    id: uuid.UUID = Field(default_factory=uuid.uuid4, description="The unique identifier of the document.", primary_key=True)
```

Comment:
a thing to think about - I've added stripe ids (strings) to the ORM, with _id as the underlying uuid powering them. We don't need to do this - I can remove them and go back to a plain uuid as the id for every object if you want. This is what stripe ids look like: 

```python
>> doc = Document(**some_values)
>> doc.id
'document-9454ea28-802e-4d7a-ba8a-863955fb7f3a'
>> doc._id
<UUID '9454ea28-802e-4d7a-ba8a-863955fb7f3a'>
```
The benefit is you can always clearly see if you have the right type of identifier. HUGE for debugging customer issues when you look at their API call and they are sending `persona-9454ea28-802e-4d7a-ba8a-863955fb7f3a` to the `get_agent` endpoint :) 

but it does mean all the schemas need to handle them. This is how I've done that in the past: 

```python
def id_regex_pattern(prefix: str):
    """ generates the regex pattern for a given id """
    return (
        r"^" + prefix + r"_"  # prefix string
        r"[a-fA-F0-9]{8}-"  # 8 hexadecimal characters
        r"[a-fA-F0-9]{4}-"  # 4 hexadecimal characters
        r"[a-fA-F0-9]{4}-"  # 4 hexadecimal characters
        r"[a-fA-F0-9]{4}-"  # 4 hexadecimal characters
        r"[a-fA-F0-9]{12}$"  # 12 hexadecimal characters
    )
    
class AddressSchema(RestModel):
    name: Optional[str] = ""
    company: Optional[str] = None
    address_1: str
    address_2: Optional[str] = None
    city: str
    state: StateShortcode
    urbanization: Optional[str] = None
    zipcode: str
    zipcode_ext: Optional[str] = None

class PersistedAddressSchema(AddressSchema):
    id: str = Field(
        ..., examples=["addr_1234-5678-9012-3456"], pattern=id_regex_pattern("addr")
    )
```

(so you know there's no magic, `RestModel` is the schema base class that does the camelCasing we've talked about elsewhere): 

```python
class RestModel(BaseModel):
    model_config = ConfigDict(
        # allows you to use the snake or camelcase names in your code (ie user_id or userId)
        populate_by_name=True,
        # allows you do dump a sqlalchemy object directly (ie PersistedAddress.model_validate(SQLAdress)
        from_attributes=True,
        # This uses pyhumps.case to change names to camelCase when writing to the API 
        alias_generator=case,
        # throw errors if attributes are given that don't belong
        extra="forbid",
    )
```

since `.id` is the stripe id on the ORM you wouldn't need to do anything fancy - just set it like a normal attribute when you are consuming a database representation